lib/resources/azure/azure_network_security_group.rb

```diff
@@ -0,0 +1,117 @@
+# encoding: utf-8
+
+require 'resources/azure/azure_backend'
+
+module Inspec::Resources
+  class AzureNetworkSecurityGroup < AzureResourceBase
+    name 'azure_network_security_group'
+
+    desc '
+      Inspec Resource to test Azure Network Security Groups (NSG)
```

Comment:
Make this a single line and add could you add an `example` showing basic usage?

Reply:
Thank you @miah I've added usage examples and made the description one line.